instagram/src/App.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+
+import Authenticate from './components/Authentication/Authenticate';
+import PostsPage from './components/PostsPage/PostsPage'
+import dummyData from './dummy-data';
+
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      filter: '',
+      commentText: [],
+      logedInUser: localStorage.getItem('user'),
+    }
+  }
+
+  logOut = () => {
+    localStorage.setItem('user', 'nope')
+    this.props.logOutTop()
+    this.setState({
+      logedInUser : ''
+    });
+  }
+  componentDidMount(){
+    this.setState({
+     data: dummyData 
+    })
+  };
+
+  handleInputChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+      });
+
+  };
+
+  handleInputChangeComment = event => {
+    let copy = this.state.commentText.slice()
+    copy[event.target.name] = event.target.value
+    
+    this.setState({
+      commentText : copy
+    });
+  };
+
+  addNewComment = (event, index) => {
```

Comment:
This entire series of functions is such a great solution I love how you managed to keep everything up here. 

Is there some reason why you opted to create a comment array like this rather than alter the data in your comments arrays within your main array of posts objects?